### PR TITLE
refactor: remove optimize C7 test dependencies

### PR DESCRIPTION
**Pull request by RomanJRW**
## Description

Initial PR to remove the remaining C7 test dependencies (excluding some of the commented out files, such as the extension).

I would like pipelines to be tested with this before merging to make sure it won't break things like the ES and OS tests, and others using the docker-compose files I am deleting here

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Comment by misiekhardcore** [comment]
@RomanJRW I added one more commit with more cleanup

File: .github/workflows/optimize-deploy-artifacts.yml [hunk 1]
```diff
@@ -76,15 +76,15 @@ jobs:
       with:
         compose_file: .github/actions/compose/docker-compose.smoketest.yml
         project_name: smoketest
+        timeout: 600
```
Comment (misiekhardcore) [inline .github/workflows/optimize-deploy-artifacts.yml:79]: we need to reduce this timeout to something reasonable

File: .github/workflows/optimize-deploy-artifacts.yml [hunk 2]
```diff
@@ -76,15 +76,15 @@ jobs:
...
       env:
-        OPTIMIZE_IMAGE_TAG: ${{ env.DOCKER_TAG }}
+        OPTIMIZE_IMAGE_TAG: ${{ env.RELEASE_VERSION }}
```
Comment (misiekhardcore) [inline .github/workflows/optimize-deploy-artifacts.yml:81]: why do we do this change here?

File: .github/workflows/optimize-deploy-artifacts.yml (hunk 1 above)
Comment (misiekhardcore) [reply .github/workflows/optimize-deploy-artifacts.yml:80]: actually we can remove this param here completely, the default value of 300s works fine

**Review by matthewBearCamunda — CHANGES_REQUESTED** [review]
The changes LGTM, however I did find references to h2 and postgres in `optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/EngineDatabaseExtension.java.out`

Does that file, and possibly the others in the same directory, need adjusting too?

**Comment by RomanJRW** [comment]
@matthewBearCamunda - good question. Ultimately, we are likely to delete all the IT things because of their C7-centricity, including this extension (this is even Engine specific). So it will be deleted in a separate PR. For the moment, the code is commented out and it has no impact here

**Review by matthewBearCamunda — APPROVED** [review]
Makes sense, approved!

**Pull request merged**